skbase/lookup/_lookup.py

```diff
@@ -110,7 +109,13 @@ def _is_non_public_module(module_name: str) -> bool:
     is_non_public : bool
         Whether the module is non-public or not.
     """
-    is_non_public: bool = "._" in module_name
+    if not isinstance(module_name, str):
+        raise ValueError(
+            f"Parameter `module_name` should be str, but found {type(module_name)}."
+        )
+    is_non_public: bool = "._" in module_name or (
+        "." not in module_name and module_name.startswith("_")
```

Comment:
should this not simply be
`"._" in module_name or module_name.startswith("_")`?

Because the current condition does not cover a module without submodule starting with `_`.

Reply:
@fkiraly, that's right, it is simpler and will accomplish the same thing.